app/controllers/api/v1/jwt_api/questionnaire_controller.rb

```diff
@@ -5,13 +5,19 @@ module V1
     module JwtApi
       class QuestionnaireController < JwtApiController
         before_action :check_admin_authenticated, only: %i[create]
-        before_action :set_questionnaire, only: %i[show]
+        before_action :check_distributions_enabled, only: %i[distribution]
+        before_action :set_questionnaire, only: %i[show distribution]
+        before_action :set_distribution_and_check_min_responses, only: %i[distribution]
 
         def show
           # TODO: Add different formats
           render json: @questionnaire, serializer: Api::QuestionnaireSerializer
         end
 
+        def distribution
```

Comment:
maybe move to a results controller?

Reply:
perhaps as a subcontroller somehow; `questionnaire/1/results/distribution` or something

Reply:
Fixed.